sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/RowBasedHashMapGenerator.scala

```diff
@@ -48,6 +48,8 @@ class RowBasedHashMapGenerator(
     val keySchema = ctx.addReferenceObj("keySchemaTerm", groupingKeySchema)
     val valueSchema = ctx.addReferenceObj("valueSchemaTerm", bufferSchema)
 
+    val numVarLenFields = groupingKeys.map(_.dataType).count(dt => !UnsafeRow.isFixedLength(dt))
```

Comment:
super nit: `.count(!UnsafeRow.isFixedLength(_))`?

Reply:
plz keep the comment `// TODO: consider large decimal and interval type` below

Reply:
@cloud-fan We want to discuss, how to modify?

Reply:
@cloud-fan We want to discuss, how to modify?

Reply:
the code style doesn't matter, both are fine. but let's keep the comment.